src/drivers/net/virtio/mod.rs

```diff
@@ -65,8 +65,13 @@ impl RxQueues {
 		//
 		let packet_size = if dev_cfg.features.contains(virtio::net::F::MRG_RXBUF) {
 			1514
-		} else {
+		} else if dev_cfg.features.contains(virtio::net::F::GUEST_TSO4)
+			|| dev_cfg.features.contains(virtio::net::F::GUEST_TSO6)
+			|| dev_cfg.features.contains(virtio::net::F::GUEST_UFO)
+		{
 			dev_cfg.raw.as_ptr().mtu().read().to_ne().into()
+		} else {
+			1514
```

Comment:
I think this branch will cause an error when we have MTU negotiated. 

> If the driver negotiates VIRTIO_NET_F_MTU, it MUST supply enough receive buffers to receive at least one receive packet of size mtu (plus low level ethernet header length) with gso_type NONE or ECN.

I don't know what MTU the device reports if guest segmentation offload is negotiated, but I think the safe behaviour is to use a constant 65550 (not the mtu) as the buffer size when it is negotiated, and if not, to use the MTU.  

([VIRTIO Specification v1.2, § 5.1.4.2](https://docs.oasis-open.org/virtio/virtio/v1.2/cs01/virtio-v1.2-cs01.html#x1-2250002))